packages/ui-testable/src/testable.ts

```diff
@@ -34,45 +35,57 @@ const testable =
   // ALWAYS_APPEND_UI_TESTABLE_LOCATORS=1
   // We do this because adding those `data-cid` locators slows things down.
   !process.env.ALWAYS_APPEND_UI_TESTABLE_LOCATORS
-    ? () => (Component) => Component
-    : decorator((ComposedComponent) => {
+    ? () => (Component: any) => Component
+    : decorator((ComposedComponent: any) => {
         const displayName =
           ComposedComponent.displayName || ComposedComponent.name
         const locator = {
           attribute: 'data-cid',
           value: displayName
         }
         const selector = `[${locator.attribute}~="${locator.value}"]`
-        class TestableComponent extends ComposedComponent {
+        class TestableComponent<
+          P = Record<string, unknown>,
+          S = ComponentState
+        > extends ComposedComponent {
           static selector = selector
-          componentDidMount(...args) {
+          private _testableUnmounted: boolean | undefined
+          _locatorTimeout: NodeJS.Timeout | undefined
```

Comment:
This is a tricky one. Although NodeJS.timeout fits here, its wrong since this package is used in the browser. Ive followed the advice here to solve it: https://stackoverflow.com/questions/51040703/what-return-type-should-be-used-for-settimeout-in-typescript in debounce: https://github.com/instructure/instructure-ui/blob/master/packages/debounce/src/debounce.ts#L71

Reply:
Done, thanks